etsd/core/views.py

```diff
@@ -20,30 +20,17 @@ def get_form_kwargs(self):
 
     def form_valid(self,form):
         usrs = form.cleaned_data['users']
-        auth = self.request.user.get_authority()
+        auth = form.instance
         initial_users = get_user_model().objects.filter(authorities=auth)
         added_users = set(usrs).difference(set(initial_users))
         removed_users = set(initial_users).difference(set(usrs))
-        for usr in added_users:
-            if usr.authorities.exists():
-                messages.add_message(
-                    self.request, 
-                    messages.ERROR, 
-                    _("""User {0} belongs to authority {1} and cannot be added! 
-                      Remove {0} from {1} and try again.""".format(usr.username, usr.authorities.first().name))
-                )
-                return HttpResponseRedirect(self.request.path)
 
         for usr in added_users:
-            usr.user_permissions.add(Permission.objects.get(codename='change_authority'))
-            usr.user_permissions.add(Permission.objects.get(codename='view_authority'))
-            usr.user_permissions.add(Permission.objects.get(codename='user'))
+            if not usr.has_perm('core.admin'):
+                usr.user_permissions.add(5)
```

Comment:
What's that 5 here ??

Reply:
Permission's 'core.user' id. Got stuck finding a workaround to that.

Reply:
This won't work; the id of the permissions may not be always the same. Use something like this:

`Permission.objects.get(codename="user", content_type__model="globalpermissionholder")`

for the core.user permission. You need to define both the name of the permission and the name of the model.